xpsi/likelihoods/_poisson_likelihood_given_background.pyx

```diff
@@ -198,8 +198,20 @@ def poisson_likelihood_given_background(double exposure_time,
     for i in range(<size_t>STAR.shape[0]):
         for j in range(<size_t>STAR.shape[1]):
             EXPEC = (STAR[i,j] + background[i,j]/n) * exposure_time
-            LOGLIKE -= EXPEC
-            LOGLIKE += counts[i,j] * log(EXPEC)
+
+            # Ensuring that the log likelihood doesn't crash
+            # even when model counts is zero in a given phase bin.
+            if(EXPEC > 0.0):
+                LOGLIKE -= EXPEC
+                LOGLIKE += counts[i,j] * log(EXPEC)
+    
+            elif(counts[i,j] == 0 and EXPEC == 0):
+                pass
+            
+            # Penalizing situations where the data counts are not zero 
+            # but the model counts are.
+            else:
+                LOGLIKE = -1.0e90 * (0.1 + 0.9*np.random.rand(1))
```

Comment:
Just a small comment, but would it perhaps make more sense to have 
`LOGLIKE += -1.0e90 * (0.1 + 0.9*np.random.rand(1))`?

Currently, it will add log-likelihoods only from those bins that happen to come after this condition, but from those that are before.

Or break out from the loop when having this condition so that likelihoods from other phase-energy bins won't get added on top of this (although then the star and background signals are not saved for all bins)?

Reply:
Yea, I think adding it to `LOGLIKE` should be okay. I'll add that change.